hashedindex/textparser.py

```diff
@@ -58,9 +58,26 @@ def normalize_unicode(text):
         return text
 
 
+def match_tokens(text, tokenize_whitespace):
+    first = True
+    for token in re.findall(_re_token, text):
+        if tokenize_whitespace and not first:
+            yield ' '
```

Comment:
Most of the changes present in this PR are to enable this `yield` statement; they're required so that the rest of the code supports consuming from [`matched_tokens`](https://github.com/MichaelAquilina/hashedindex/pull/17/files#diff-bca7df3b022757454ef625ad3579a5b5R118) as a generator.

Alternatively it would've been possible to add the whitespace tokens in-place into the list returned by [`re.findall`](https://github.com/MichaelAquilina/hashedindex/pull/17/files#diff-bca7df3b022757454ef625ad3579a5b5R63), as suggested in [this Stackoverflow thread](https://stackoverflow.com/questions/5920643/add-an-item-between-each-item-already-in-the-list).

Reply:
:grey_question: This implementation doesn't appear to be correct - this will emit whitespace tokens between a word token and a punctuation token.

Reply:
A better solution may be to allow the token regex to explictly match groups of whitespace as a token, and then emit one space character for each of those.